### DISCUSSION TITLE
Can I reduce brew's timeout when accessing the Internet?

### DISCUSSION THREAD

**Discussion by halloleo** [Homebrew/discussions#2948] [Linux] [answered]
On one CentOS machine with Linuxbrew I'm not always connected to the internet, but I still would like to search for formulas in my local store.

When I do so,  `brew` tries first to search on GitHub via `curl`, fails after a loooong time and then lists what I want to know: the local information.

How can I speed up this process? Can I tell brew to fail faster?

**Comment by gromgit** [comment, marked as answer]
`export HOMEBREW_NO_GITHUB_API=1` to let `brew` know not to search online.
```
$ man brew
[...]
ENVIRONMENT
       Note that environment variables must have a value set to be detected. For example, run export HOMEBREW_NO_INSECURE_REDIRECT=1 rather than just export HOMEBREW_NO_INSECURE_REDIRECT.
[...]
       HOMEBREW_NO_GITHUB_API
              If set, do not use the GitHub API, e.g. for searches or fetching relevant issues after a failed install.
```
